components/FilterForm/FilterForm.js

```diff
@@ -1,10 +1,40 @@
-import React from "react";
+import React, { useState } from "react";
+import FilteredCarsList from "./FilteredCarsList";
+import data from "../../data.json";
 
 function FilterForm() {
+  const [list, setList] = useState(data);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    //fetch input data
+    const formData = new FormData(event.target);
+    const inputData = Object.fromEntries(formData);
+    //destructing
+    const { CountryOfManufacture, BodyType, Fuel, minPrice, maxPrice } =
+      inputData;
+
+    //if the number is in the range
+    function between(x, min, max) {
+      return x >= min && x <= max;
+    }
+
+    setList(
+      data.filter(
+        (car) =>
+          car.CountryOfManufacture === `${CountryOfManufacture}` &&
```

Comment:
Since `CountryOfManufacture` is of type `string`, you don't need to wrap it in a template ^^

Same for all the other occurrences.

Wrapping the prices in `Number(),` is good. Observe how the check works also without the cast operation, because the comparison operators (`>=`, `<=`, etc...) coerce the passed strings to their number representations before being applied.

Reply:
Cool! Really works! Thanks a lot!!

Reply:
@moonwave99 May I give further to QA now?